src/SourceFile.js

```diff
@@ -20,14 +20,20 @@ export class SourceFile {
   encodedLiterals = new Map();
   warnings = [];
 
-  constructor(sourceCodeString) {
+  constructor(sourceCodeString, customProbes = []) {
     this.tracer = new VariableTracer()
       .enableDefaultTracing()
       .trace("crypto.createHash", {
         followConsecutiveAssignment: true, moduleName: "crypto"
       });
 
-    this.probesRunner = new ProbeRunner(this);
+    if (Array.isArray(customProbes) && customProbes.length > 0) {
+      this.probesRunner = new ProbeRunner(this, customProbes);
```

Comment:
```suggestion
      this.probesRunner = new ProbeRunner(this, [...ProbeRunner.Defaults, ...customProbes]);
```

Reply:
We need to keep default probes (maybe we could introduce an option to choose to keep them or not). But for me by default we extend probes (and not overwriting them)

Reply:
agreed this would give more flexibility/control, and help passing the tests easier as well (when we inject one probe). 
What do you have in mind for that option ?
- append (default) : add custom probes to the list 
- replace : replace list of default probe by the custom one

Reply:
Yep that how I see it right now. Outside of that there is probably long term subject like "do we want to inject the probe at the beginning, at the end, or before a given probe ?". But I guess we can deal with that later :)